internal/providers/testing/provider_mock.go

```diff
@@ -632,11 +632,6 @@ func (p *MockProvider) RenewEphemeralResource(r providers.RenewEphemeralResource
 		return resp
 	}
 
-	if p.CloseEphemeralResourceCalled {
-		resp.Diagnostics = resp.Diagnostics.Append(fmt.Errorf("CloseEphemeralResource called on %q before RenewEphemeralResource", r.TypeName))
-		return resp
-	}
-
```

Comment:
It does, just not consistently. It was put in there by mistake for a specific test, but checking the order of calls in the mock doesn't really work because multiple ephemeral resources may be calling renew and close at different time.